### PR TITLE
Use shadow helium api

### PR DESCRIPTION
**Pull request by posterzh**
**Issue**

- Link: 
https://github.com/NebraLtd/hm-config/issues/195
https://github.com/NebraLtd/maker-starter-app/issues/143

- Summary:
 Use a private Helium API to get the Hotpot details reliably.

**How**

**Screenshots**

**References**

**Checklist**

- [ ] Tests added
- [ ] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [ ] Thought about variable and method names

### CODE REVIEW

File: src/utils/appDataClient.ts [hunk 1]
```diff
@@ -51,17 +136,43 @@ export const hotspotOnChain = async (address: string) => {
...
+  } catch (error) {
+    try {
+      hotspotActivityList = await heliumHttpClient
+        .hotspot(gatewayAddress)
+        .activity.list({
+          filterTypes: ['poc_receipts_v1', 'poc_request_v1'],
+        })
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the hotspot activity list: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:171]: Helium API request failed while trying to retrieve hotspot {gatewayAddress} activity.

File: src/utils/appDataClient.ts [hunk 2]
```diff
@@ -51,17 +136,43 @@ export const hotspotOnChain = async (address: string) => {
...
 export const getChainVars = async (keys?: string[]) => {
-  return heliumHttpClient.vars.get(keys)
+  try {
+    return privateHeliumClient.vars.get(keys)
+  } catch (error) {
+    try {
+      return heliumHttpClient.vars.get(keys)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the chain vars: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:146]: Helium API request failed while trying to retrieve chain variables for  {keys}.

File: src/utils/appDataClient.ts [hunk 3]
```diff
@@ -1,44 +1,129 @@
...
 export const getPredictedOraclePrice = async () => {
-  return heliumHttpClient.oracle.getPredictedPrice()
+  try {
+    return privateHeliumClient.oracle.getPredictedPrice()
+  } catch (error) {
+    try {
+      return heliumHttpClient.oracle.getPredictedPrice()
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the predicted oracle price: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:123]: Helium API request failed while trying to retrieve the predicted oracle price.

File: src/utils/appDataClient.ts [hunk 4]
```diff
@@ -1,44 +1,129 @@
...
 export const getCurrentOraclePrice = async () => {
-  return heliumHttpClient.oracle.getCurrentPrice()
+  try {
+    return privateHeliumClient.oracle.getCurrentPrice()
+  } catch (error) {
+    try {
+      return heliumHttpClient.oracle.getCurrentPrice()
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the oracle price: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:109]: Helium API request failed while trying to retrieve the current oracle price.

File: src/utils/appDataClient.ts [hunk 5]
```diff
@@ -1,44 +1,129 @@
...
+  try {
+    const { data } = await privateHeliumClient.accounts.get(address)
+    return data
+  } catch (error) {
+    try {
+      const { data } = await heliumHttpClient.accounts.get(address)
+      return data
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the account information: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:81]: Helium API request failed while trying to retrieve account information for {address}.

File: src/utils/appDataClient.ts [hunk 6]
```diff
@@ -1,44 +1,129 @@
...
+    try {
+      const newHotspotList = await heliumHttpClient.accounts
+        .fromAddress(address)
+        .hotspots.list()
 
-  return newHotspotList.takeJSON(10000)
+      return newHotspotList.takeJSON(10000)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the list of hotspots: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:63]: Helium API request failed while trying to retrieve the list of hotspots belonging to {address}.

File: src/utils/appDataClient.ts [hunk 7]
```diff
@@ -1,44 +1,129 @@
...
+    }
+  }
+}
+
+export const submitTxn = async (txn: string) => {
+  try {
+    return await heliumHttpClient.transactions.submit(txn)
+  } catch (err) {
+    throw new Error(
+      `Helium API call is failed while submitting the transaction: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:39]: Helium API request failed while submitting the transaction:

File: src/utils/appDataClient.ts [hunk 8]
```diff
@@ -1,44 +1,129 @@
...
 export const getHotspotDetails = async (address: string): Promise<Hotspot> => {
-  return heliumHttpClient.hotspots.get(address)
+  try {
+    return await privateHeliumClient.hotspots.get(address)
+  } catch (error) {
+    try {
+      return await heliumHttpClient.hotspots.get(address)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the hotspot details: ${err}`,
```
Comment (marvinmarnold) [inline src/utils/appDataClient.ts:28]: Helium API request failed while trying to retrieve hotspots for address {address}

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Will these error messages be displayed to the user? I guess this is the follow up ticket to fix that behavior? https://github.com/NebraLtd/maker-starter-app/issues/143

**Comment by posterzh** [comment]
> Will these error messages be displayed to the user? I guess this is the follow up ticket to fix that behavior? #143

Right

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**